### PR TITLE
Keyboard nav and aria improvements

### PR DESCRIPTION
**Pull request by limscoder**
#### What does this PR do?
keyboard nav:
 * navigate between weeks with up/down arrow keys when inner grid is focused
 * navigate between years with up/down arrow keys when outer div is focused

aria:
 * added additional aria labels so screen reader reads date when focus changes (tested on OSX/chrome/voiceover and win8/IE11/JAWS)

#### Any background context you want to provide?
There are 5 commits. Started with some refactoring to simplify DayPicker, then added keyboard nav features and aria labels to date cells, so that it works better with screen readers.

#### Where should the reviewer start?
Look at each commit independently.

#### How should this be manually tested?
Test keyboard nav in example app.
Test with screen reader in example app (cmd-f5 to enable voiceover in OSX).

### CODE REVIEW

**Comment by gpbl** [comment]
Hi @limscoder thanks so much for this PR! Really appreciated. I'm on vacation now, I can give a look to it in the next days. 🌴

**Comment by limscoder** [comment]
Any progress on reviewing this? Can I answer any questions?

File: src/DayPicker.js [hunk 1]
```diff
@@ -429,10 +468,17 @@ export default class DayPicker extends Component {
         tabIndex = this.props.tabIndex;
       }
     }
+
+    const ariaLabel = this.props.localeUtils.formatDate ?
+      this.props.localeUtils.formatDate(day) : day.toDateString();
```
Comment (gpbl) [inline src/DayPicker.js]: The `formatDate` here is a new function to add to the [LocaleUtils](http://www.gpbl.org/react-day-picker/docs/LocaleUtils.html) right? I can take care of this (need to update docs, tests, etc. :smile:) 
We should add the `locale` props as second argument to the function, so implementers can use their own localization utils.

Comment (limscoder) [inline src/DayPicker.js]: yep, new function.

**Comment by gpbl** [comment]
Thanks again @limscoder, those are great additions. They are going in the next major release, I'm working on it this week.

**Comment by limscoder** [comment]
thx

**Pull request merged**